engine/access/state_stream/backend/event_retriever.go

```diff
@@ -0,0 +1,88 @@
+package backend
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/rs/zerolog"
+
+	"github.com/onflow/flow-go/engine/access/index"
+	"github.com/onflow/flow-go/model/flow"
+	"github.com/onflow/flow-go/storage"
+	"github.com/onflow/flow-go/utils/logging"
+)
+
+// EventsResponse represents the response containing events for a specific block.
+type EventsResponse struct {
+	BlockID flow.Identifier
+	Height  uint64
+	Events  flow.EventsList
+}
+
+// EventsRetriever retrieves events related to blocks
```

Comment:
```suggestion
// EventsRetriever retrieves events by block height. It can be configured to retrieve events from the events indexer(if available)
// or using a dedicated callback to query it from other sources.
```